dockstore-webservice/pom.xml

```diff
@@ -919,7 +871,11 @@
                             <failOnWarning>true</failOnWarning>
                             <outputXML>true</outputXML>
                             <ignoreNonCompile>true</ignoreNonCompile>
-
+                            <ignoredDependencies>
```

Comment:
Why did these get added? A comment could be useful.

Reply:
*1. Just to understand, is this BOM supposed to have only the common dependencies between dockstore-cli and dockstore-webservice? Or is it between dockstore-cli and the whole dockstore repo?*
This BOM contains all the dependencies for both the dockstore-cli and the dockstore repo projects. This is because the original dependencyManagement sections are located in the top level parent POM, and the dependecyManagement section is inherited by the other POMs that indicate the top level POM as their parent.

*2. I'm not wild about the name dockstore-projects-bom -- I would lean towards dockstore-bom, or depending on the answer to above, dockstore-common-bom.*
I can change the name to dockstore-bom but the BOM is also used in dockstore-cli which is a separate repo, and the dockstore repo, and so dockstore-bom seems to not indicate that it covers the dockstore-cli. The name dockstore-common also might indicate it is for the dockstore-common project only.

*Shouldn't io.dockstore.common.dockstore-common be in the bom? That is a shared Dockstore dependency between the web service and the CLI. There are also a few others, but the only one you have is io.dockstore.zenodo-client-generated. If on the other hand, those others shouldn't be in there, then why is zenodo-client-generated in the bom?*
As I understand it, the dependencyManagement section provides version information to the other project's <dependencies> sections. (https://maven.apache.org/pom.html#Distribution_Management) So in the BOM the element
```
    <dependencyManagement>
        <dependencies>
...
            <dependency>
                <groupId>io.dockstore</groupId>
                <artifactId>zenodo-client-generated</artifactId>
                <version>0.2</version>
            </dependency>
...
```
Specifies that in the POMs that inherit the parent BOM  ( e.g. dockstore-event-consumer/pom.xml), in the <dependencies> section, the <dependency> element zenodo-client-generated must use version .2  
```
<dependencies>
...
        <dependency>
            <groupId>io.dockstore</groupId>
            <artifactId>zenodo-client-generated</artifactId>
        </dependency>
 

```
~/Documents/UCSC/gitroot/dockstore => grep -rni --include=pom.xml 'zenodo-client-generated' *
dockstore-event-consumer/pom.xml:42:            <artifactId>zenodo-client-generated</artifactId>
dockstore-projects-bom/pom.xml:69:                <artifactId>zenodo-client-generated</artifactId>
```